service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -162,50 +145,7 @@ boolean hasEhrStatusReceivedAckWithUnexpectedConditionErrors(String conversation
         return ehrReceivedAckErrorDetailsThatContainsSearchedErrCodeAndMsg.isPresent();
     }
 
-    void updateEhrExtractStatusListWithEhrReceivedAcknowledgementError(Stream<EhrExtractStatus> ehrExtractStatusList,
-                                                                       String errorCode,
-                                                                       String errorMessage) {
-
-        ehrExtractStatusList.forEach(ehrExtractStatus -> {
-            try {
-                updateEhrExtractStatusWithEhrReceivedAckError(ehrExtractStatus.getConversationId(), errorCode, errorMessage);
-            } catch (EhrExtractException exception) {
-
-                logger().error("An error occurred when updating EHR Extract with Ack erorrs, EHR Extract Status conversation_id: {}",
-                               ehrExtractStatus.getConversationId(), exception);
-                throw exception;
-            } catch (Exception exception) {
-                logger().error("An unexpected error occurred for conversation_id: {}", ehrExtractStatus.getConversationId(), exception);
-                throw exception;
-            }
-        });
-    }
-
-    private void updateEhrExtractStatusWithEhrReceivedAckError(String conversationId,
-                                                               String errorCode,
-                                                               String errorMessage) {
-
-        Update update = createUpdateWithUpdatedAt();
-        update.addToSet(RECEIVED_ACK_ERRORS, ErrorDetails.builder().code(errorCode).display(errorMessage).build());
-
-        EhrExtractStatus ehrExtractStatus = mongoTemplate.findAndModify(
-            createQueryForConversationId(conversationId),
-            update,
-            getReturningUpdatedRecordOption(),
-            EhrExtractStatus.class);
-
-        if (ehrExtractStatus == null) {
-            throw new EhrExtractException(format(
-                "Couldn't update EHR received acknowledgement with error information because EHR status doesn't exist, conversation_id: %s",
-                conversationId));
-        }
-
-        logger().info("EHR status (EHR received acknowledgement) record successfully "
-                      + "updated in the database with error information conversation_id: {}", conversationId);
-
-    }
```

Comment:
So I think the boundary we want is between code whos reason to change is due to MongoDB, and code whos reason to change is due to the business requirments of timing out transfers that haven't been acknowledged within a set amount of time.

_I think_ the code in this class should have the responsibility for MongoDB. _I also think_ that the code you've removed here is also responsible for marshaling acknowledgement data into MongoDB. If you also agree, then I'd keep them together.

Reply:
ok, yes, it would make sense. I will move it back to EhrExtractStatusService class

Reply:
yes